src/SHCore.Desktop/SHCore.cs

```diff
@@ -0,0 +1,103 @@
+    /// Exported functions from the SHCore.dll Windows library
+    /// that are available to Desktop apps only.
+    /// </content>
+    public static partial class SHCore
+    {
+        /// <summary>
+        /// Queries the dots per inch (dpi) of a display.
+        /// The values of <paramref name="dpiX"/> and <paramref name="dpiY"/> are identical.
+        /// You only need to record one of the values to determine the DPI and respond appropriately.
+        /// </summary>
+        /// <param name="hmonitor">Handle of the monitor being queried</param>
+        /// <param name="dpiType">The type of DPI being queried. Possible values are from the <see cref="MONITOR_DPI_TYPE"/> enumeration</param>
+        /// <param name="dpiX">The value of the DPI along the X axis. This value always refers to the horizontal edge, even when the screen is rotated</param>
+        /// <param name="dpiY">The value of the DPI along the Y axis. This value always refers to the vertical edge, even when the screen is rotated</param>
+        /// <returns>
+        /// This function returns one of the following values:
+        /// <list>
+        /// <item><see cref="HResult.Code.S_OK"/>: The function successfully returns the X and Y DPI values for the specified monitor</item>
+        /// <item><see cref="HResult.Code.E_INVALIDARG"/>: The handle, DPI type, or pointers passed in are not valid</item>
+        /// </list>
+        /// </returns>
+        /// <remarks>
+        /// <para>
+        /// When <see cref="MONITOR_DPI_TYPE"/> is <see cref="MONITOR_DPI_TYPE.MDT_ANGULAR_DPI"/> or <see cref="MONITOR_DPI_TYPE.MDT_RAW_DPI"/>,
+        /// the returned DPI value does not include any changes that the user made to the DPI by using the desktop scaling override slider control in Control Panel.
+        /// </para>
+        /// <para>
+        /// When you call <see cref="GetDpiForMonitor"/>, you will receive different DPI values depending on the DPI awareness of the calling application.
+        /// DPI awareness is an application-level property usually defined in the application manifest.
+        /// The following table indicates how the results will differ based on the <see cref="PROCESS_DPI_AWARENESS"/> value of your application.
+        /// </para>
+        /// <list>
+        /// <item><see cref="PROCESS_DPI_AWARENESS.PROCESS_DPI_UNAWARE"/>: 96 because the app is unaware of any other scale factors</item>
+        /// <item><see cref="PROCESS_DPI_AWARENESS.PROCESS_SYSTEM_DPI_AWARE"/>:A value set to the system DPI because the app assumes all applications use the system DPI</item>
+        /// <item><see cref="PROCESS_DPI_AWARENESS.PROCESS_PER_MONITOR_DPI_AWARE"/>: The actual DPI value set by the user for that display</item>
+        /// </list>
+        /// </remarks>
+        [DllImport(nameof(SHCore), SetLastError = true)]
+        public static extern HResult GetDpiForMonitor(IntPtr hmonitor, MONITOR_DPI_TYPE dpiType, out int dpiX, out int dpiY);
```

Comment:
OK, I will transform that on an issue. Some handles may not have ways to Close or Destroy or it may not make sense to do that on the API, on these cases we could have something like [HandleRef ](https://msdn.microsoft.com/en-us/library/system.runtime.interopservices.handleref(v=vs.110).aspx) or a descendant of that.

MSDN Says:
> If you use platform invoke to call a managed object, and the object is not referenced elsewhere after the platform invoke call, it is possible for the garbage collector to finalize the managed object. This action releases the resource and invalidates the handle, causing the platform invoke call to fail. Wrapping a handle with HandleRef guarantees that the managed object is not garbage collected until the platform invoke call completes.

> The HandleRef value type, like GCHandle, is a special type recognized by the interop marshaler. A normal, nonpinned GCHandle also prevents untimely garbage collection, yet HandleRef provides better performance. Although using HandleRef to keep an object alive for the duration of a platform invoke call is preferred, you can also use the GC.KeepAlive method for the same purpose.